File: spring/dining-service/src/main/java/fr/univcotedazur/dining/controllers/dto/TableWithOrderDTO.java
package fr.univcotedazur.dining.controllers.dto;

import javax.validation.constraints.Positive;
import java.util.UUID;

public class TableWithOrderDTO {

    @Positive
    private Long number;

    private boolean taken;

    private UUID tableOrderId; // could be null (is null if taken == false)
    // could also be complemented with more information on the order itself, here as a MVP, we only add the id
    // forcing for a callback to the API to get the information

    public Long getNumber() {
        return number;
    }

    public void setNumber(Long number) {
        this.number = number;
    }

    public boolean isTaken() {
        return taken;
    }

    public void setTaken(boolean taken) {
        this.taken = taken;
    }

    public UUID getTableOrderId() {
        return tableOrderId;
    }

    public void setTableOrderId(UUID tableOrderId) {
        this.tableOrderId = tableOrderId;
    }

}

File: nestjs/dining-service/src/tables-with-order/dto/table-with-order.dto.ts
import { IsBoolean, IsMongoId, IsNotEmpty, IsOptional, IsPositive } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

import { TableOrder } from '../../table-orders/schemas/table-order.schema';
import { Table } from '../../tables/schemas/table.schema';

export class TableWithOrderDto {
  @ApiProperty()
  @IsMongoId()
  _id: string;

  @ApiProperty()
  @IsNotEmpty()
  @IsPositive()
  number: number;

  @ApiProperty()
  @IsNotEmpty()
  @IsBoolean()
  taken: boolean;

  @ApiProperty()
  @IsMongoId()
  @IsOptional()
  tableOrderId: string; // could be null (is null if taken === false)
  // could also be complemented with more information on the order itself, here as a MVP, we only add the id
  // forcing for a callback to the API to get the information

  static tableWithOrderDtoFactory(table: Table, tableOrder: TableOrder): TableWithOrderDto {
    const tableWithOrderDto: TableWithOrderDto = new TableWithOrderDto();
    tableWithOrderDto._id = table._id;
    tableWithOrderDto.number = table.number;
    tableWithOrderDto.taken = table.taken;
    tableWithOrderDto.tableOrderId = tableOrder?._id || null;

    return tableWithOrderDto;
  }
}
